src/components/Card.js

```diff
@@ -1,16 +1,41 @@
 export default class Card {
-  constructor({ name, link, handleImageClick }, cardSelector) {
+  constructor(
+    { id, isLiked, name, link, handleImageClick, handleRemoveCardClick, handleLikeIcon },
+    cardSelector
+  ) {
     //cardSelector= #card-template
+    this._id = id;
     this._name = name;
     this._link = link;
+    this._isLiked = isLiked;
     this._cardSelector = cardSelector; //template
     this._handleImageClick = handleImageClick;
+    this._handleRemoveCardClick = handleRemoveCardClick;
+    this._handleLikeIcon = handleLikeIcon;
   }
+
+  getId() {
+    return this._id;
+  }
+
+  setLikeStatus(status) {
+    this._isLiked = status;
+
+  }
+
+  getLikeStatus() {
+    return this._isLiked;
+  }
+
   _setEventListeners() {
     this._cardElement
       .querySelector(".card__like-button")
       .addEventListener("click", () => {
-        this._handleLikeIcon();
+        this._handleLikeIcon(this).then(()=>{
+          this._cardElement
+          .querySelector(".card__like-button")
+          .classList.toggle("card__like-button_active");
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.

Here you use `then` but... it goes after `catch` which is in `index.js`, so this toggling will work even after an error

Please, move this code into `setLikeStatus` and call this method in `then`